tests/system/src/testapp7/java/mymod/myapp7/LocalStorageAccessWithModuleLayer.java

```diff
@@ -0,0 +1,99 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package myapp7;
+
+import java.lang.module.ModuleDescriptor;
+import javafx.application.Application;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.concurrent.Worker.State;
+import javafx.scene.Scene;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+
+public class LocalStorageAccessWithModuleLayer extends Application {
+    public static final int ERROR_OK = 0;
+    public static final int ERROR_ASSUMPTION_VIOLATED = 2;
+    public static final int ERROR_TIMEOUT = 3;
+    public static final int ERROR_TITLE_NOT_UPDATED = 4;
+    public static final int ERROR_UNEXPECTED_EXIT = 5;
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        Module module = Application.class.getModule();
+
+        if (module == null) {
+            System.err.println("Failure: Module for Application not found");
+            System.exit(ERROR_ASSUMPTION_VIOLATED);
+        }
+
+        if (! module.isNamed()) {
+            System.err.println("Failure: Expected named module");
+            System.exit(ERROR_ASSUMPTION_VIOLATED);
+        }
+
+        ModuleDescriptor moduleDesc = module.getDescriptor();
+
+        if (moduleDesc.isAutomatic()) {
+            System.err.println("Failure: Automatic module found");
+            System.exit(ERROR_ASSUMPTION_VIOLATED);
+        }
+
+        if (moduleDesc.isOpen()) {
+            System.err.println("Failure: Open module found");
+            System.exit(ERROR_ASSUMPTION_VIOLATED);
+        }
+
+        BorderPane root = new BorderPane();
+        Scene scene = new Scene(root);
+
+        WebView webview = new WebView();
+        root.setCenter(webview);
+
+        // The loaded test page LocalStorageAccess will set the title to
+        // 'Executed'. This indicates, that the second script block was reached
+        webview.getEngine().getLoadWorker().stateProperty().addListener(
+                new ChangeListener<State>() {
+                    public void changed(ObservableValue ov, State oldState, State newState) {
+                        if (newState == State.SUCCEEDED) {
+                            String title = webview.getEngine().getTitle();
+                            if ("Executed".equals(title)) {
+                                System.exit(ERROR_OK);
+                            } else {
+                                System.exit(ERROR_TITLE_NOT_UPDATED);
+                            }
+                        }
+                    }
+                });
+        webview.getEngine().load(LocalStorageAccessWithModuleLayer.class.getResource("/LocalStorageAccess.html").toExternalForm());
```

Comment:
This fails to find the resource when I run the test via gradle. I recommend putting it in the same package as the class and removing the leading `/` (e.g., see the FXML resources in `testapp6` or `testscriptapp1`).

Reply:
Can you tell me how you ran it? I ensured, that the test failed with a segfault before continuing with the fix. While I did that I indeed placed the file in the wrong directory, but currently I see no test failures locally. I run:

```
./gradlew  -PFULL_TEST=true -PUSE_ROBOT=true :systemTests:cleanTest :systemTests:test --tests test.com.sun.webkit.LocalStorageAccessTest 
```

to only invoke the new test. I'll move the file to the package, but I'd like to understand where I went wrong.

Reply:
That's how I ran it, too. I did it from a clean repo. This was on Windows, but shouldn't matter.

Reply:
I just tried it on Linux and get the same failure:

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.net.URL.toExternalForm()" because the return value of "java.lang.Class.getResource(String)" is null
        at mymod/myapp7.LocalStorageAccessWithModuleLayer.start(LocalStorageAccessWithModuleLayer.java:92)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:849)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:474)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:447)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:446)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
        at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
        at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
        ... 1 more
```

Reply:
I get the same error. For some reason, the resource isn't found when it's on the root.
Moving it inside the app7 package work.

Reply:
Thank you both for checking. I noticed, there were two copies of the HTML present in the `build` folder, which explains the strange behavior.